File: app/src/main/java/com/laurelkeys/userinformationtestingrobots/userinformation/UserInformationMVP.java
package com.laurelkeys.userinformationtestingrobots.userinformation;

import android.support.annotation.NonNull;

import com.laurelkeys.userinformationtestingrobots.userinformation.model.UserInformation;

/**
 * Created by Tiago in July 2016.
 */
public interface UserInformationMVP {

    interface Model {
        boolean isValidName(String name);

        boolean isValidPhoneNumber(String phoneNumber);

        boolean isValidEmail(String email);

        boolean isValidAge(String age);
    }

    interface View {
        void displaySuccess();

        void displayFailure();
    }

    interface Presenter {
        void onSendClick(@NonNull UserInformation userInformation);
    }
}

File: app/src/main/java/com/laurelkeys/userinformationtestingrobots/userinformation/UserInformationMVP.kt
package com.laurelkeys.userinformationtestingrobots.userinformation

import com.laurelkeys.userinformationtestingrobots.userinformation.model.UserInformation

/**
 * Created by Tiago in July 2016.
 */
interface UserInformationMVP {

    interface Model {
        fun isValidName(name: String): Boolean

        fun isValidPhoneNumber(phoneNumber: String?): Boolean

        fun isValidEmail(email: String): Boolean

        fun isValidAge(age: String): Boolean
    }

    interface View {
        fun displaySuccess()

        fun displayFailure()
    }

    interface Presenter {
        fun onSendClick(userInformation: UserInformation)
    }
}
